crates/host/src/wasmbus/mod.rs

```diff
@@ -3944,6 +3961,18 @@ fn injector_to_headers(injector: &TraceContextInjector) -> async_nats::header::H
         .collect()
 }
 
+/// Helper function to ensure an individual instance's annotations contain all the
+/// filter annotations, determining if an instance matches a filter.
+///
+/// See #607 for more details.
+fn annotations_match_filter(annotations: &Annotations, filter: &Annotations) -> bool {
+    filter.iter().all(|(k, v)| {
+        annotations
+            .get(k)
+            .is_some_and(|instance_value| instance_value == v)
```

Comment:
nit; couldn't we do something like 
```suggestion
        annotations.get(k).as_deref() == Some(v)
```
instead?

Reply:
We could, feels like personal preference and I don't mind either way